#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material1 "casterMat"
      coulombFriction [
        0
      ]
    }
  ]
}
Viewpoint {
  orientation 0.9832475184565501 0.16607698332167697 0.075118260495086 5.420880927453806
  position 0.7289112963686541 2.3219336650029665 1.7148939086879882
}
TexturedBackground {
  luminosity 0.16
}
TexturedBackgroundLight {
  luminosity 0.16
}
Floor {
  size 2.45 4.9
  appearance Parquetry {
    type "dark strip"
    colorOverride 0 0 0
  }
}
DEF key_cylinder Solid {
  translation 0.015324 0.0249804 -0.462378
  rotation 0.707106781394287 0.7071067809765698 -1.77919267514899e-06 3.1415785211699707
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 0.9
        metalness 0
      }
      geometry Cylinder {
        height 0.05
        radius 0.025
      }
    }
  ]
  name "solid(key2)(1)"
  boundingObject Cylinder {
    height 0.05
    radius 0.025
  }
  physics Physics {
    density -1
    mass 0.05
  }
}
DEF key_cylinder Solid {
  translation 0.84 0.0249804 -0.59
  rotation 0.5773502691896258 0.5773502691896258 0.5773502691896258 2.0944
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0.666667 0
        roughness 0.9
        metalness 0
      }
      geometry Cylinder {
        height 0.05
        radius 0.025
      }
    }
  ]
  name "solid(key2)"
  boundingObject Cylinder {
    height 0.05
    radius 0.025
  }
  physics Physics {
    density -1
    mass 0.05
  }
}
DEF key_box Solid {
  translation 0.085087 0.0249902 -0.163264
  rotation 2.3804048428874433e-17 -1 2.8427918894088784e-17 0.785402
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 0.9
        metalness 0
      }
      geometry Box {
        size 0.05 0.05 0.05
      }
    }
  ]
  name "solid(key1)"
  boundingObject Box {
    size 0.05 0.05 0.05
  }
  physics Physics {
    density -1
    mass 0.05
  }
}
DEF key_box Solid {
  translation 0.43 0.0249902 -0.66
  rotation 0.08962309529846825 0.7735535142806973 -0.6273616670893313 2.9730453408535576e-17
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0.666667 0
        roughness 0.9
        metalness 0
      }
      geometry Box {
        size 0.05 0.05 0.05
      }
    }
  ]
  name "solid(key1)(1)"
  boundingObject Box {
    size 0.05 0.05 0.05
  }
  physics Physics {
    density -1
    mass 0.05
  }
}
DEF Light SpotLight {
  attenuation 0 0 1
  beamWidth 1.5
  cutOffAngle 1.57
  direction 0 -1 0
  intensity 4
  location 0 2 0
  radius 10
  castShadows TRUE
}
DEF Arena_maze Solid {
  translation 1.225 0 0.59
  rotation 0 1 0 -1.5707996938995747
  children [
    DEF WALLS Shape {
      appearance PBRAppearance {
        roughness 0.9
        metalness 0
      }
      geometry Mesh {
        url [
          "maze_test_arena.STL"
        ]
      }
    }
  ]
  name "WALLS"
  boundingObject USE WALLS
}
DEF Arena_CMYK Solid {
  translation 1.225 0 -0.93
  rotation 0 1 0 -1.5707996938995747
  children [
    DEF CMYK_walls Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 0.9
        metalness 0
      }
      geometry Mesh {
        url [
          "CMYK_area_fillet.STL"
        ]
      }
    }
  ]
  name "WALLS(CMYK)"
  boundingObject USE CMYK_walls
}
DEF Arena_track1 Solid {
  translation -0.625 0 -0.01
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF TRACK Shape {
      appearance PBRAppearance {
        roughness 0.9
        metalness 0
      }
      geometry Mesh {
        url [
          "entrance_path_test_arena.STL"
        ]
      }
    }
  ]
  name "TRACK(entrance)"
}
DEF Arena_CMYK_floor Solid {
  translation 0.755 0 -0.47
  children [
    DEF TF_C Transform {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            roughness 0.9
            metalness 0
          }
          geometry Box {
            size 0.92 0.001 0.9
          }
        }
      ]
    }
    DEF TF_Y Transform {
      translation -0.92 0 0.75
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            roughness 0.9
            metalness 0
          }
          geometry Box {
            size 0.92 0.001 0.6
          }
        }
      ]
    }
    DEF TF_M Transform {
      translation 0 0 0.75
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.9
            metalness 0
          }
          geometry Box {
            size 0.92 0.001 0.6
          }
        }
      ]
    }
    DEF Key_hole_outline Transform {
      translation -1.38 0.07 1.05
      rotation -0.5773509358554485 -0.5773509358554485 0.5773489358556708 -2.094395307179586
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                roughness 0.9
                metalness 0
              }
              geometry Mesh {
                url [
                  "key_hole_outline.STL"
                ]
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(CMYK_floor)"
}
DEF Ping_pong1 Solid {
  translation 0 0.019997881039999995 -0.3
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 0.9
        metalness 0
      }
      geometry Sphere {
        radius 0.02
        subdivision 2
      }
    }
  ]
  name "solid(ppong1)"
  boundingObject Sphere {
    radius 0.02
    subdivision 2
  }
  physics Physics {
    density -1
    mass 0.0027
  }
}
DEF Ping_pong2 Solid {
  translation -0.46 0.019997881039999995 -0.62
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 0.9
        metalness 0
      }
      geometry Sphere {
        radius 0.02
        subdivision 2
      }
    }
  ]
  name "solid(ppong2)"
  boundingObject Sphere {
    radius 0.02
    subdivision 2
  }
  physics Physics {
    density -1
    mass 0.0027
  }
}
DEF plexRobot Robot {
  translation 0.731116 0.0311016 0.0435983
  rotation -0.0005616232027299765 -0.99999836097092 0.0017212306213152191 3.1375853071795863
  children [
    Display {
      children [
        Shape {
          appearance Appearance {
            material Material {
              emissiveColor 1 1 1
            }
            texture ImageTexture {
              filtering 0
            }
          }
        }
      ]
      width 128
      height 128
    }
    Camera {
      translation -3.18636e-05 0.101049 0.0994359
      rotation 0 -0.9636078869569907 -0.2673197340158101 3.14012
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.005
          }
        }
      ]
      name "cam"
      fieldOfView 1.4
      width 128
      height 128
    }
    DEF cameraPoll Solid {
      translation 0 0.06 0.09
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.02 0.12 0.01
          }
        }
      ]
      name "solid(5)"
    }
    DEF handle HingeJoint {
      jointParameters HingeJointParameters {
        position -3.1864964000760665e-09
        anchor 0 0 0.1125
      }
      device [
        RotationalMotor {
          name "handleMotor"
        }
        PositionSensor {
          name "handleEncoder"
        }
      ]
      endPoint Solid {
        translation 0 0 0.1125
        rotation 1 0 0 0
        children [
          DEF leftSlider SliderJoint {
            jointParameters JointParameters {
              position -4.919436679505517e-07
              axis -1 0 0
            }
            device [
              PositionSensor {
                name "leftSliderEncoder"
              }
              LinearMotor {
                name "leftSlider"
              }
            ]
            endPoint Solid {
              translation 0.0675004919415815 -0.01 0.0425
              children [
                DEF leftTouchSensor TouchSensor {
                  translation -0.003000017450736303 -5.319595149794187e-09 0.0019999796188978625
                  rotation 0.5721969731630641 0.24775053713214737 0.7817993957875384 -1.3521334550386928e-15
                  children [
                    DEF leftSensor Shape {
                      appearance PBRAppearance {
                        baseColor 0 0 0
                        roughness 1
                        metalness 0
                      }
                      geometry Box {
                        size 0.001 0.01 0.065
                      }
                    }
                  ]
                  name "leftTouchSensor"
                  boundingObject USE leftSensor
                  physics Physics {
                  }
                }
                DEF leftSlider Shape {
                  appearance PBRAppearance {
                    baseColor 1 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.005 0.025 0.07
                  }
                }
              ]
              name "solid(2)"
              boundingObject USE leftSlider
              physics Physics {
              }
            }
          }
          DEF rightSlider SliderJoint {
            jointParameters JointParameters {
              position -4.920725972078578e-07
              axis 1 0 0
            }
            device [
              PositionSensor {
                name "rightSliderEncoder"
              }
              LinearMotor {
                name "rightSlider"
              }
            ]
            endPoint Solid {
              translation -0.06750049207468366 -0.01 0.0425
              children [
                DEF rightTouchSensor TouchSensor {
                  translation 0.003000017450164094 -5.150490405697594e-09 0.001999979615703973
                  rotation 0.8222979524755927 0.21734408852889645 -0.5259159861955055 -1.2715641234386462e-15
                  children [
                    DEF rightSensor Shape {
                      appearance PBRAppearance {
                        baseColor 0 0 0
                        roughness 1
                        metalness 0
                      }
                      geometry Box {
                        size 0.001 0.01 0.065
                      }
                    }
                  ]
                  name "rightTouchSensor"
                  boundingObject USE rightSensor
                  physics Physics {
                  }
                }
                DEF rightSlider Shape {
                  appearance PBRAppearance {
                    baseColor 1 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.005 0.025 0.07
                  }
                }
              ]
              name "solid(4)"
              boundingObject USE rightSlider
              physics Physics {
              }
            }
          }
          DEF rightHandle Solid {
            translation -0.045 -0.01 0.005
            children [
              DEF rightHandle Shape {
                appearance PBRAppearance {
                  baseColor 0 0 1
                  roughness 1
                  metalness 0
                }
                geometry Box {
                  size 0.05 0.025 0.005
                }
              }
            ]
            name "solid(1)"
            boundingObject USE rightHandle
            physics Physics {
              density -1
              mass 0.04
            }
          }
          DEF leftHandle Solid {
            translation 0.045 -0.01 0.005
            children [
              DEF leftHandle Shape {
                appearance PBRAppearance {
                  baseColor 0 0 1
                  roughness 1
                  metalness 0
                }
                geometry Box {
                  size 0.05 0.025 0.005
                }
              }
            ]
            name "solid(8)"
            boundingObject USE leftHandle
            physics Physics {
              density -1
              mass 0.04
            }
          }
          DEF handle Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.14 0.01 0.005
            }
          }
        ]
        name "solid(4)"
        boundingObject USE handle
        physics Physics {
          density -1
          mass 0.5
        }
      }
    }
    DEF basePlate Group {
      children [
        Transform {
          translation 0 0 0.005
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0 0 1
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.08 0.005 0.07
              }
            }
          ]
        }
        Transform {
          translation 0 0 0.075
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0 1 1
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.14 0.005 0.07
              }
            }
          ]
        }
      ]
    }
    DEF leftWheel HingeJoint {
      jointParameters HingeJointParameters {
        position 4.889600043336328
        anchor 0.057 0 0
      }
      device [
        RotationalMotor {
          name "leftMotor"
        }
      ]
      endPoint Solid {
        translation 0.057 -1.953918560380442e-07 1.0910278706315754e-06
        rotation 0.5404094721914363 0.5399793009042403 0.6452751017672548 4.288411356205894
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.025
              radius 0.03
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF rightWheel HingeJoint {
      jointParameters HingeJointParameters {
        position 3.3503995675636506
        anchor -0.057 0 0
      }
      device [
        RotationalMotor {
          name "rightMotor"
        }
      ]
      endPoint Solid {
        translation -0.057 1.0843107880017577e-06 2.2976054257362286e-07
        rotation 0.7054560192359282 -0.7048944692641583 -0.07386198023745538 3.289165893806314
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.025
              radius 0.03
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF leftCaster Solid {
      translation 0.055 -0.01625 0.09
      children [
        DEF caster Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.01375
          }
        }
      ]
      name "solid(2)"
      contactMaterial "casterMat"
      boundingObject USE caster
      physics Physics {
      }
    }
    DEF rightCaster Solid {
      translation -0.055 -0.01625 0.09
      rotation 0 0 1 1.57
      children [
        DEF caster Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.01375
          }
        }
      ]
      name "solid(3)"
      contactMaterial "casterMat"
      boundingObject USE caster
      physics Physics {
      }
    }
  ]
  name "robot(1)"
  boundingObject USE basePlate
  physics Physics {
  }
  controller "testing_controller"
}
